src/packet/one_pass_signature.rs

```diff
@@ -32,14 +34,14 @@ impl OnePassSignature {
     pub fn from_slice(packet_version: Version, input: &[u8]) -> Result<Self> {
         let (_, pk) = parse(packet_version)(input)?;
 
-        if pk.version != 2 && pk.version != 3 && pk.version != 4 && pk.version != 5 {
+        if pk.version != 2 && pk.version != 3 && pk.version != 4 && pk.version != 6 {
             unsupported_err!("unsupported signature version {}", pk.version);
         }
 
         Ok(pk)
     }
 
-    pub fn from_details(
+    pub fn from_details_v3(
```

Comment:
don't we need a `v4`?

Reply:
OnePassSignature is only defined for v3 and v6

Reply:
right.. that is not confusing at all..

Reply:
right, we definitely need more documentation pointing people into the RFC.
i haven't spent a lot of time thinking about that, yet.